### DISCUSSION TITLE
C API: Callback for progress reporting

### DISCUSSION THREAD

**Discussion by Giorgi** [duckdb/duckdb#14022] [Ideas]
It is possible to get a progress report by polling `duckdb_query_progress` but would be great if we could register a callback and get notified of progress changes.

I peeked at the [source code](https://github.com/duckdb/duckdb/blob/d697acfb108f6ec1b1ed26f0062445e1d49ee1c4/src/main/client_context.cpp#L427) and it looks like there are only four places where `query_progress` is manipulated so if my understanding is correct you would need to invoke the callback only in those four places.

Similar to https://github.com/duckdb/duckdb/discussions/11923

**Comment by ashokgudur** [comment]
When I use DBeaver Database Manager it is able to show the progress of long running processes like data import and export.
It looks like the DuckDB Java JDBC driver implemented that API.

I want similar thing for ODBC. Need to look at the C API.
